cpp/include/experimental/detail/graph_utils.cuh

```diff
@@ -51,51 +51,65 @@ rmm::device_uvector<edge_t> compute_major_degree(
   rmm::device_uvector<edge_t> degrees(0, handle.get_stream());
 
   vertex_t max_num_local_degrees{0};
-  for (int i = 0; i < col_comm_size; ++i) {
-    auto vertex_partition_idx =
-      partition.is_hypergraph_partitioned()
-        ? static_cast<size_t>(row_comm_size) * static_cast<size_t>(i) +
-            static_cast<size_t>(row_comm_rank)
-        : static_cast<size_t>(col_comm_size) * static_cast<size_t>(row_comm_rank) +
-            static_cast<size_t>(i);
+  for (int i = 0; i < (partition.is_hypergraph_partitioned() ? col_comm_size : row_comm_size);
```

Comment:
@aschaffer - Note that reviewing this is a mess due to the nature of how this PR currently contains the contents of multiple PRs that haven't been merged into the baseline yet.

All of your comments on the following files should be redirected to PR 1174 where these changes come from so that @seunghwak can address them there:
 * `cpp/include/experimental/detail/graph_utils.cuh`
 * `cpp/include/patterns/*`
 * `cpp/include/utilities/comm_utils.cuh`
 * `cpp/src/experimental/graph*.cu`

Reply:
Okay. Feel free to resolve / disregard.